posthog/queries/test/test_trends.py

```diff
@@ -0,0 +1,977 @@
+        with freeze_time(freeze_args[2]):
+            Event.objects.create(
+                team=self.team,
+                event="sign up",
+                distinct_id="blabla",
+                properties={"$some_property": "other_value", "$some_numerical_prop": 80,},
+            )
+            Event.objects.create(team=self.team, event="no events", distinct_id="blabla")
+
+            # second team should have no effect
+            Event.objects.create(
+                team=secondTeam, event="sign up", distinct_id="blabla", properties={"$some_property": "other_value"},
+            )
+        return sign_up_action, person
+
+    def _create_breakdown_events(self):
+        freeze_without_time = ["2020-01-02"]
+
+        sign_up_action = Action.objects.create(team=self.team, name="sign up")
+        ActionStep.objects.create(action=sign_up_action, event="sign up")
+
+        with freeze_time(freeze_without_time[0]):
+            for i in range(25):
+                Event.objects.create(
+                    team=self.team, event="sign up", distinct_id="blabla", properties={"$some_property": i},
+                )
+
+    def _compare_entity_response(self, response1, response2, remove=("action", "label")):
+        if len(response1):
+            for attr in remove:
+                response1[0].pop(attr)
+        else:
+            return False
+        if len(response2):
+            for attr in remove:
+                response2[0].pop(attr)
+        else:
+            return False
+        return str(response1[0]) == str(response2[0])
+
+    def test_trends_per_day(self):
+        self._create_events()
+        with freeze_time("2020-01-04T13:00:01Z"):
+            # with self.assertNumQueries(16):
+            response = Trends().run(
+                Filter(data={"date_from": "-7d", "events": [{"id": "sign up"}, {"id": "no events"}],}), self.team
+            )
+        self.assertEqual(response[0]["label"], "sign up")
+        self.assertEqual(response[0]["labels"][4], "Wed. 1 January")
+        self.assertEqual(response[0]["data"][4], 3.0)
+        self.assertEqual(response[0]["labels"][5], "Thu. 2 January")
+        self.assertEqual(response[0]["data"][5], 1.0)
+
+    def test_trends_per_day_48hours(self):
+        self._create_events()
+        with freeze_time("2020-01-03T13:00:01Z"):
+            action_response = self.client.get("/api/action/trends/?date_from=-48h&interval=day").json()
+            event_response = self.client.get(
+                "/api/action/trends/",
+                data={
+                    "date_from": "-48h",
+                    "events": jdumps([{"id": "sign up"}, {"id": "no events"}]),
+                    "interval": "day",
+                },
+            ).json()
+
+        self.assertEqual(action_response[0]["data"][1], 1.0)
+        self.assertEqual(action_response[0]["labels"][1], "Thu. 2 January")
+
+        self.assertTrue(self._compare_entity_response(action_response, event_response))
```

Comment:
@EDsCODE I might remove all of these comparisons and just the actions functionality once to simplify and speed up tests. Thoughts? I'm not super sure what they were originally supposed to test

Reply:
the compare entity response was meant to make sure the responses for actions and event responses don't differ if the entities are the same

Reply:
Yep, i've left most of those in noww